File: app/src/main/java/eu/sesma/dagger/platform/MainActivityModule.java
package eu.sesma.dagger.platform;

import eu.sesma.dagger.di.PerActivity;
import eu.sesma.dagger.ui.main.MainActivity;

import android.app.Activity;

import dagger.Module;
import dagger.Provides;

@Module
public abstract class MainActivityModule {
    @Provides
    @PerActivity
    static Activity provideActivity(MainActivity activity) {
        return activity;
    }
}

File: module_main/src/main/java/eu/sesma/dagger/main/di/MainActivityModule.kt
package eu.sesma.dagger.main.di

import eu.sesma.dagger.core.di.ActivityProvider
import eu.sesma.dagger.main.*
import org.koin.android.ext.koin.androidContext
import org.koin.core.qualifier.named
import org.koin.dsl.module

val mainActivityModule = module {
    scope(named<MainActivity>()) {

        factory { MainCollaborator() }

        scoped { MainScopedCollaborator() }

        scoped {
            MainNavigator(
                    context = androidContext(),
                    coreCollaborator = get(),
                    coreSingletonCollaborator = get(),
                    mainCollaborator = get(),
                    mainScopedCollaborator = get(),
                    activity = (androidContext() as ActivityProvider).activeActivity
            )
        }

        scoped {
            MainPresenter(
                    navigator = get(),
                    mainCollaborator = get(),
                    mainScopedCollaborator = get(),
                    coreSingletonCollaborator = get(),
                    coreCollaborator = get()
            )
        }
    }
}